fluss-common/src/main/java/org/apache/fluss/utils/ArrowUtils.java

```diff
@@ -434,9 +450,24 @@ private static Field toArrowField(String fieldName, DataType logicalType) {
             for (DataField field : rowType.getFields()) {
                 children.add(toArrowField(field.getName(), field.getType()));
             }
+        } else if (logicalType instanceof MapType) {
+            MapType mapType = (MapType) logicalType;
+            DataType keyType = mapType.getKeyType();
+            if (keyType.isNullable()) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Map key type must be non-nullable for Arrow conversion, but got: %s",
+                                keyType.asSummaryString()));
+            }
+            Field keyField = toArrowField("key", keyType);
+            Field valueField = toArrowField("value", mapType.getValueType());
+            FieldType structFieldType = new FieldType(false, ArrowType.Struct.INSTANCE, null);
+            List<Field> structChildren = new ArrayList<>();
+            structChildren.add(keyField);
+            structChildren.add(valueField);
+            Field structField = new Field("entries", structFieldType, structChildren);
```

Comment:
Use `MapVector.KEY_NAME`, `MapVector.VALUE_NAME` and `MapVector.DATA_VECTOR_NAME` to replace the hard code string names ("key", "value", "entires").